lucid/src/Lucid/Htmx/Extra.hs

```diff
@@ -0,0 +1,202 @@
+{- |
+Module      : Lucid.Htmx.Extra
+Description : Provides extra htmx tags
+
+This module defines additional attributes that can be used to get additional
+behaviour
+<https://htmx.org/reference/#attributes-additional>
+-}
+module Lucid.Htmx.Extra where
+
+import Data.Foldable
+import Data.List (intersperse)
+import Data.Text (Text, pack)
+import Lucid (Html, HtmlT, script_, src_)
+import Lucid.Base (Attribute, makeAttribute)
+import Lucid.Htmx.Extension
+import Lucid.Htmx.Render
+
+-- | <https://htmx.org/attributes/hx-boost/>
+-- add progressive enhancement for links and forms
+hxBoost_ :: Text -> Attribute
+hxBoost_ = makeAttribute "hx-boost"
+
+-- | <https://htmx.org/attributes/hx-confirm/>
+-- shows a confirm() dialog before issuing a request
+hxConfirm_ :: Text -> Attribute
+hxConfirm_ = makeAttribute "hx-confirm"
+
+-- | <https://htmx.org/attributes/hx-delete/>
+-- issues a DELETE to the specified URL
+hxDelete_ :: Text -> Attribute
+hxDelete_ = makeAttribute "hx-delete"
+
+-- | <https://htmx.org/attributes/hx-disable/>
+-- disables htmx processing for the given node and any children nodes
+hxDisable_ :: Attribute
+hxDisable_ = makeAttribute "hx-disable" mempty
+
+-- | <https://htmx.org/attributes/hx-disabled-elt/>
+-- adds the disabled attribute to the specified elements while a request is in flight
+hxDisabledElt_ :: Text -> Attribute
+hxDisabledElt_ = makeAttribute "hx-disabled-elt"
+
+-- | <https://htmx.org/attributes/hx-disinherit/>
+-- control and disable automatic attribute inheritance for child nodes
+hxDisinherit_ :: Text -> Attribute
+hxDisinherit_ = makeAttribute "hx-disinherit"
+
+-- | <https://htmx.org/attributes/hx-encoding/>
+-- changes the request encoding type
+hxEncoding_ :: Text -> Attribute
+hxEncoding_ = makeAttribute "hx-encoding"
+
+-- | <https://htmx.org/attributes/hx-ext/>
+-- extensions to use for this element
+hxExt_ :: Text -> Attribute
+hxExt_ = makeAttribute "hx-ext"
+
+-- | A typesafe version of 'hxExt_' that works with the "included" extensions
+-- that the htmx codebase is tested against
+hxExtension_ :: HtmxExtension -> Attribute
+hxExtension_ = makeAttribute "hx-ext" . render
+
+-- | Include multiple extensions in one declaration
+hxExtensions_ :: [HtmxExtension] -> Attribute
+hxExtensions_ = makeAttribute "hx-ext" . fold . intersperse "," . fmap render
```

Comment:
These are new, and have a typed interface. I didn't want to break backwards compat, so gave them new names.